test/test_dht.cpp

```diff
@@ -186,15 +186,78 @@ void send_simple_dht_request(node& node, char const* msg, udp::endpoint const& e
 	}
 }
 
+void write_peers(entry::dictionary_type& r, std::set<tcp::endpoint> const& peers)
+{
+	entry::list_type& pe = r["values"].list();
+	for (std::set<tcp::endpoint>::const_iterator it = peers.begin()
+		 ; it != peers.end(); ++it)
+	{
+		std::string endpoint(18, '\0');
```

Comment:
Almost. Because write_endpoint takes the output iterator by non-const reference you have to manually instantiate a local std::back_insert_iterator<std::string> and pass it in.

Suddenly it becomes clear why stl algorithms generally take output iterators by value and return an updated copy :)